colossalai/zero/sharded_param/sharded_param.py

```diff
@@ -1,44 +1,38 @@
+from typing import Optional, Tuple, Union
+
 import numpy
 import torch
 import torch.distributed as dist
 from colossalai.context.parallel_mode import ParallelMode
 from colossalai.core import global_context as gpc
 from colossalai.zero.sharded_model._zero3_utils import get_shard
 from colossalai.zero.sharded_param import ShardedTensor
-from typing import Union, Tuple, Optional
 
 
 class ShardedParamV2(object):
 
     def __init__(self, param: torch.nn.Parameter, process_group: Optional[dist.ProcessGroup] = None) -> None:
-        self._data_sharded_tensor = ShardedTensor(param.data, process_group)
-        if param.requires_grad and param.grad is not None:
-            self._grad_sharded_tensor = ShardedTensor(param.grad, process_group)
-            param.grad = None
-        else:
-            self._grad_sharded_tensor = None
+        self._data_sharded_tensor: ShardedTensor = ShardedTensor(param.data, process_group)
+        self._grad_sharded_tensor: Optional[torch.Tensor] = None
 
         # make sure the shared param is the only owner of payload
         param.data = torch.empty([], dtype=param.dtype, device=param.device)
 
     @property
     def data(self):
-        return self._data_sharded_tensor.payload
-
-    @data.setter
-    def data(self, t: torch.Tensor):
-        self._data_sharded_tensor.payload = t
+        return self._data_sharded_tensor
```

Comment:
zero context init should be updated accordingly.

Reply:
Ok, I will fix this.